pd/src/app.rs

```diff
@@ -102,14 +107,14 @@ impl App {
         genesis_block.set_height(0);
 
         // Note that errors cannot be handled in InitChain, the application must crash.
-        let genesis: Vec<GenesisNote> = serde_json::from_slice(&init_chain.app_state_bytes)
+        let genesis: GenesisAppState = serde_json::from_slice(&init_chain.app_state_bytes)
```

Comment:
Non-blocking: I think it would be better to use bincode here, since we already use bincode for the note commitment tree, but it's not a big deal either way.